src/StreamContextFactory.php

```diff
@@ -30,13 +31,29 @@ public function create(ConfigInterface $config)
             if (isset($proxy['proxy_login']) && isset($proxy['proxy_password'])) {
                 // Support for proxy authentication is untested. The current implementation is based on
                 // http://php.net/manual/en/function.stream-context-create.php#74431.
-                $options['http']['header'] =  array(
-                    'Proxy-Authorization: Basic ' .
-                    base64_encode($proxy['proxy_login'] . ':' . $proxy['proxy_password'])
-                );
+                $headers[] = 'Proxy-Authorization: Basic ' .
+                    base64_encode($proxy['proxy_login'] . ':' . $proxy['proxy_password']);
             }
         }
 
+        // Native SoapClient uses basic authentication by defaults,
+        // and `authentication` option can be omitted.
+        $soapOptions = $config->get('soapClientOptions') + array(
+            'authentication' => SOAP_AUTHENTICATION_BASIC
+        );
```

Comment:
I must admit that I do not understand the comment and the logic behind this block.

Why is it necessary to set `SOAP_AUTHENTICATION_BASIC` as the default authentication method before the subsequent tests?